CMakeLists.txt

```
@@ -1213,13 +1213,25 @@ endif()
 if(CONFIG_OUTPUT_DISASSEMBLY)
   set(out_disassembly_cmd    "")
   set(out_disassembly_byprod "")
-  bintools_objdump(
-    RESULT_CMD_LIST    out_disassembly_cmd
-    RESULT_BYPROD_LIST out_disassembly_byprod
-    DISASSEMBLE_SOURCE
-    FILE_INPUT         ${KERNEL_ELF_NAME}
-    FILE_OUTPUT        ${KERNEL_LST_NAME}
-    )
+
+  if(CONFIG_OUTPUT_DISASSEMBLE_ALL)
+    bintools_objdump(
+      RESULT_CMD_LIST    out_disassembly_cmd
+      RESULT_BYPROD_LIST out_disassembly_byprod
+      DISASSEMBLE_ALL
+      FILE_INPUT         ${KERNEL_ELF_NAME}
+      FILE_OUTPUT        ${KERNEL_LST_NAME}
+      )
+  else()
+    bintools_objdump(
+      RESULT_CMD_LIST    out_disassembly_cmd
+      RESULT_BYPROD_LIST out_disassembly_byprod
+      DISASSEMBLE_SOURCE
+      FILE_INPUT         ${KERNEL_ELF_NAME}
+      FILE_OUTPUT        ${KERNEL_LST_NAME}
+      )
+  endif()
+
```

No need to supplicate the call to `bintools_objdump`
```suggestion
  set(disassemble_type DISASSEMBLE_SOURCE)
  if(CONFIG_OUTPUT_DISASSEMBLE_ALL)
    set(disassemble_type DISASSEMBLE_ALL)
  endif()
    
  bintools_objdump(
    RESULT_CMD_LIST    out_disassembly_cmd
    RESULT_BYPROD_LIST out_disassembly_byprod
    ${disassemble_type}
    FILE_INPUT         ${KERNEL_ELF_NAME}
    FILE_OUTPUT        ${KERNEL_LST_NAME}
    )
```

---

Done!